### PR TITLE
Track bundles in the new unified format

### PR DESCRIPTION
**Pull request by lcarva**
Ref: EC-383

### CODE REVIEW

File: internal/tracker/tracker.go [hunk 1]
```diff
@@ -239,14 +292,75 @@ func filterRecords(records []bundleRecord, prune bool) []bundleRecord {
...
+				ExpiresOn:   task.ExpiresOn,
+			})
+		}
+	}
+
+	for _, bundles := range t.TaskBundles {
+		// Sort the task bundles in reverse order. The first bundle being the most recent one. The
+		// sorting function returns true if the bundle at "i" is considered newer than the bundle at
+		// "j". It is assumed that every bundle has an EffectiveOn date and a Tag, but some bundles
+		// may not have an ExpiresOn date.
```
Comment (simonbaird) [inline internal/tracker/tracker.go:332]: Nice commentary.

**Review by simonbaird — APPROVED** [review]
I might be misremembering the plan, but are we still intending to phase out the "expires_on" field from the trusted tasks data? If so then perhaps mention that in commit message and/or some commentary.

**Review by zregvart — APPROVED** [review]
Do we need the `effective_on` now?

**Comment by lcarva** [comment]
Filed EC-423 and EC-424 to deal with removing `effective_on`.

**Pull request merged**